File: Random/PruebaDeListaDoble.go
Accion ListaDoble es //la funcion recursiva que se ve aca esta en la carpeta "Python" para comprobar que funciona
 Ambiente 
	nodo = registro 
		letra: AN(1)
		prox, ant: puntero a nodo 
	freg 
	prim,p,q,ult,u,s: puntero a nodo 

	Funcion capicua(proximo,anterior: puntero a nodo): booleano 
		si (proximo = anterior) o (proximo.ant = anterior) entonces //(proximo.ant = anterior) es para el caso de que la palabra tenga cantidad par de digitos
			capicua:= verdadero 
		sino 
			si *proximo.letra = *anterior.letra entonces 
				capicua:= capicua(*proximo.prox,*anterior.ant)
			sino 
				capicua:= falso 
			fsi 
		fsi 
	FFuncion 

	procedimiento insertar() es
		Nuevo(q); *q.letra:= letra 	
		si prim = nil entonces 
			prim:= q
			ult:= q 
			*q.ant:= nil 
			*q.prox:= nil 
		sino 
			*q.prox:= nil 
			*q.ant:= ult 
			*ult.prox:= q 
			ult:= q 
		fsi 
	FProcedimiento

	letra, salir: AN 

	Proceso 
		letra:= ' '; salir:= ' '

		Esc("Presione ENTER entre cada letra, presione 0 y ENTER para terminar")
		repetir 
			Esc("Ingrese una letra: "); Leer(letra)
			insertar()	
		hasta que (Letra = "0")

		p:= prim; u:= ult
		si capicua(p,u) entonces 
			Esc("La palabra es capicua")
		sino 
			Esc("La palabra no es capicua")
		fsi 
	FProceso 
FAccion 

File: Python/PruebaDeListaDoble.py
class Nodo:
    def __init__(self, letra):
        self.letra = letra
        self.prox = None
        self.ant = None

class ListaDoble:
    def __init__(self):
        self.prim = None
        self.ult = None

    # Función recursiva para verificar capicúa 
    def capicua(self, proximo, anterior):
        if proximo == anterior or (proximo.ant == anterior):  # caso base
            return True
        else:
            if proximo.letra == anterior.letra:
                return self.capicua(proximo.prox, anterior.ant)
            else:
                return False

    # Procedimiento de inserción al final
    def insertar(self, letra):
        nuevo = Nodo(letra)

        if self.prim is None:  # lista vacía
            self.prim = nuevo
            self.ult = nuevo
        else:
            nuevo.ant = self.ult
            self.ult.prox = nuevo
            self.ult = nuevo

    # Método para mostrar lista (opcional, ayuda en prueba) 
    def mostrar(self):
        p = self.prim
        while p is not None:
            print(p.letra, end=" <-> ")
            p = p.prox
        print("Nil")

# --- Programa Principal ---
lista = ListaDoble()
palabra = input("Ingrese una palabra \n")

for letra in palabra:
    lista.insertar(letra)

lista.mostrar()

# Verificar si es capicúa
if lista.capicua(lista.prim,lista.ult):
    print("La palabra es capicua")
else:
    print("La palabra no es capicua")
